packages/vite-plugin-angular/src/lib/angular-vite-plugin.ts

```diff
@@ -186,10 +186,25 @@ export function angular(
 
           // return fileEmitter
           const data = typescriptResult?.content ?? '';
+          const forceAsyncTransformation =
+            /for\s+await\s*\(|async\s+function\s*\*/.test(data);
+          const useInputSourcemap = (!isProd ? undefined : false) as undefined;
+
+          if (!forceAsyncTransformation && !isProd) {
+            return {
```

Comment:
Instead of `contents` and `loader`, this should just return an object of `code` and `map`. Contents and loader are specific to esbuild transforms, and we are reusing some of that logic in Rollup transforms

```ts
{
  code: string, // typescript result content minus sourcemap string
  map: string // sourcemap from typescript if enabled
}
```

Reply:
Okay, so something like this?

```
return {
  code: data.replace(/^\/\/# sourceMappingURL=[^\r\n]*/gm, ''),
  map: useInputSourcemap ? data : null,
};
```

Reply:
Something like this should be enough

```ts
              return {
                code: !isProd
                ? data
                : data.replace(/^\/\/# sourceMappingURL=[^\r\n]*/gm, ''),
            };
```